Makefile

```diff
@@ -2,10 +2,12 @@ export V ?= 0
 
 OUTPUT_DIR := $(CURDIR)/out
 
-EXAMPLE_LIST := hello_world
-EXAMPLE_LIST += random
-EXAMPLE_LIST += aes
-EXAMPLE_LIST += hotp
+EXCLUDE_DIRS := out|docs
+EXCLUDE_CMD := perl -pe 's/($(EXCLUDE_DIRS)) //g'
+LIST_CMD := ls -d */ | cut -f1 -d'/' | awk 1 ORS=' '
+TRIM_CMD := perl -pe 's/^\s+|\s+$$//g'
+CMD := $(LIST_CMD) | $(EXCLUDE_CMD) | $(TRIM_CMD)
+EXAMPLE_LIST := $(shell $(CMD))
```

Comment:
Interesting. I find the command a bit tricky. Make script tools could help simplify ($foreach, $filter-out ...).

Reply:
Or in this case maybe something like: `EXAMPLE_LIST := $(dir $(wildcard */Makefile))` ?

Reply:
+1 re. using make functions.
But fundamentally I don't see much benefit in this change. It just makes the build more fragile (no early failure if a sub-directory is missing, failure if some unexpected directory is added to the work space).
@zaidan is there some use case you have in mind?

Reply:
@zaidan thanks for explaining your use case. It is reasonable.
The only thing I still don't like very much in the updated code is that it will happily produce an incomplete build if some source directory is missing (e.g., deleted by mistake). Not a big deal though.

Reply:
Note about the `optee_example` prefix: it is on purpose to make things clear, easier to manage (`ls /bin/optee_example*`, `bash$ optee_example<TAB><TAB>` etc.).

Reply:
@jforissier yeah, that is a valid point. We could add assertions, but this would increase the coupling. It is some kind of trade-off.

I am totally fine with the prefix, my suggestion was to replace the auto-prefixing with the convention to add the prefix explicitly. I hope this was clear.